### PR TITLE
Add more SRTP protection profiles

### PR DESCRIPTION
**Pull request by Syracusa**
Add more SRTP protection profiles from RFC8723 and RFC8269.

IANA Link : https://www.iana.org/assignments/srtp-protection/srtp-protection.xhtml

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Comment by t8m** [comment]
Unfortunately this is not something we would accept with CLA: trivial although this is probably not really copyrightable.

Could you please submit a CLA? https://www.openssl.org/policies/cla.html

**Comment by Syracusa** [comment]
> Unfortunately this is not something we would accept with CLA: trivial although this is probably not really copyrightable.
> 
> Could you please submit a CLA? https://www.openssl.org/policies/cla.html

Hi.

I've just registered at CLA DB and reopen pull request.

Could you please check?

**Review by t8m — APPROVED** [review]

File: include/openssl/srtp.h [hunk 1]
```diff
@@ -28,16 +28,28 @@
...
 
 /* AEAD SRTP protection profiles from RFC 7714 */
-# define SRTP_AEAD_AES_128_GCM  0x0007
-# define SRTP_AEAD_AES_256_GCM  0x0008
+# define SRTP_AEAD_AES_128_GCM                      0x0007
+# define SRTP_AEAD_AES_256_GCM                      0x0008
+
+/* DOUBLE AEAD SRTP protection profiles from RFC 8723 */
+# define DOUBLE_AEAD_AES_128_GCM_AEAD_AES_128_GCM   0x0009
+# define DOUBLE_AEAD_AES_256_GCM_AEAD_AES_256_GCM   0x000A
```
Comment (mattcaswell) [inline include/openssl/srtp.h:44]: There is some inconsistency here. The RFC 7714 profiles are identified in the RFC without the leading `SRTP_`, but the RFC8269 onces do have that. These new "double" protection profiles don't have the leading `SRTP_` which is kind of problematic, in that we are claiming a new namespace by not prefixing with SRTP. Should they be called `SRTP_DOUBLE_*` instead?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: include/openssl/srtp.h (hunk 1 above)
Comment (Syracusa) [reply include/openssl/srtp.h:44]: That's a reasonable opinion, should I close this pull request and make another pull request with changes? Are there something clever way to make changes without another pull request?

Comment (slontis) [reply include/openssl/srtp.h:44]: You can use
````
> git commit --fixup commit_hash
````

Get the commit_hash of any commit using
````
>git log
````

Comment (slontis) [reply include/openssl/srtp.h:44]: So you can have multiple comitts for different things and then apply fixups to any commit when being reviewed.

Comment (Syracusa) [reply include/openssl/srtp.h:44]: > 

Thanks, I'll give it a try

**Review by mattcaswell — APPROVED** [review]
Looks good. I can approve as is - but you might want to consider adding an entry to the CHANGES.md file.

**Comment by Syracusa** [comment]
> Looks good. I can approve as is - but you might want to consider adding an entry to the CHANGES.md file.

Thanks, I added entry to the CHANGES.md file.

**Comment by t8m** [comment]
You'll need to rebase your branch against the current master branch to resolve the conflict.

**Comment by Syracusa** [comment]
I rebase my commit to resolve the conflict and force-pushed the commit to remove verbose commit log.. Is it OK?

**Comment by t8m** [comment]
@mattcaswell still OK?

**Comment by Syracusa** [comment]
Removed trailing spaces form CHANGES.md

**Review by mattcaswell — APPROVED** [review]
> @mattcaswell still OK?

Still ok

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you for your contribution.

**Pull request closed without merging**